utilities/blob_db/blob_db.h

```diff
@@ -218,6 +237,14 @@ class BlobDB : public StackableDB {
 
   virtual Status SyncBlobFiles() = 0;
 
+  // Update TTL for an existing key.
+  //
+  // Caveat: The operation is not atomic. Updating a key about to expire
+  // may make the key reappear after expiration. Also it may overwrite a
+  // concurrent write.
+  virtual Status UpdateTTL(const UpdateTTLOptions& options, const Slice& key,
```

Comment:
Correct me if I interpreted this incorrectly, but it looks to me that this API does not provide much advantage as its implementation in theory is just: `UpdateTTL = Get + PutUntil`. Instead of adding a new API, why not add a new param to the `Get` API to return `expiration`, and let users handle the Update themselves by doing a Get followed by a Put?

That way, the biggest advantage is that RocksDB will not have a non-atomic API. It also has the additional advantage of allowing users to choose the type of consistency they need, i.e. if they want an atomic operation, they can take a lock themselves; if they are fine with non-atomicity, they can avoid the lock penalty.

Reply:
Make sense. My initial thought was that by having it on RocksDB side we can have an atomic API in the future. But since the current implementation is non-atomic, let's move to user side for now.